src/storage/txn/process.rs

```diff
@@ -876,6 +1055,220 @@ fn process_write_impl<S: Snapshot, L: LockMgr>(
     })
 }
 
+fn process_batch_write_impl<S: Snapshot, L: LockMgr>(
+    cid: u64,
+    cmd: Command,
+    snapshot: S,
+    lock_mgr: Option<L>,
+    statistics: &mut Statistics,
+) -> Result<BatchWriteResults> {
```

Comment:
Could we reuse some code here? There is a lot of duplicate code in `process_write_impl` and `process_batch_write_impl`. We could change it such as:
```
fn process_prewrite_impl(ctx: Context, snapshot: Snapshot, ... modifies: &mut Vec<Modify> ) -> (cid, ..) {
}
fn process_batch_write_impl() {
        .......
        match tag {
            CommandKind::batch_prewrite => {
                let len = commands.len();
                for i in 0..len {
                    if ids[i].is_none() {
                        continue;
                    }
                    let (...) = process_prewrite_impl(commands[i], &mut modifies);
                }
           }
      }
}

fn process_write_impl<S: Snapshot, L: LockMgr>(
    cmd: Command,
    snapshot: S,
    lock_mgr: Option<L>,
    statistics: &mut Statistics,
) -> Result<WriteResult> {
    let (pr, to_be_write, rows, ctx, lock_info) = match cmd {
        Command::Prewrite {
            ctx,
            mutations,
            primary,
            start_ts,
            options,
            ..
        } => {
            let (...) = process_prewrite_impl(commands[i], &mut modifies);
        }
....
     
}


```

Reply:
I don't like the look of dereferencing enum type twice in a row. Plus this only reuse a simple logic where we process `txn` results, which I think is relatively minor.